Hestia/app/src/main/java/hestia/backend/exceptions/ExceptionFactory.java

```diff
@@ -0,0 +1,41 @@
+package hestia.backend.exceptions;
+
+
+import com.google.gson.JsonObject;
+
+import org.json.JSONObject;
+
+import hestia.backend.exceptions.ServerExceptions.DatabaseException;
+import hestia.backend.exceptions.ServerExceptions.InvalidStateException;
+import hestia.backend.exceptions.ServerExceptions.NotFoundException;
+import hestia.backend.exceptions.ServerExceptions.ServerException;
+import hestia.backend.exceptions.ServerExceptions.SetupFailedException;
+import hestia.backend.exceptions.ServerExceptions.UndefinedServerException;
+
+public class ExceptionFactory {
+    JsonObject exception;
+
+    public ExceptionFactory(JsonObject exception){
+        this.exception=exception;
+    }
+
+
+    public ServerException getException(){
+        String exceptionName=exception.get("exception").getAsString();
+        JsonObject details=exception.get("details").getAsJsonObject();
+        switch(exceptionName){
+            case "DatabaseException":
+                return new DatabaseException(details.get("type").getAsString(),details.get("message").getAsString());
```

Comment:
Better to get the information from the json object in different lines to make it easier to read.